File: Codeforces/C_Email_address.py

s=input()
t=s
ans = ""


while s.find("at") != -1:
    s= s.replace("at","@")

while s.find("dot") != -1 :
    s = s.replace("dot",".")


if( s[0]=='.'):
    s=s.replace(".","dot",1)

if( s[0]=='@'):
    s=s.replace("@","at",1)

ans = s

if( s[len(s)-1]=='@'):
    ans = s[:len(s)-1] + "at"


if( s[len(s)-1]=='.'):

    ans = s[:len(s)-1] + "dot"

ans1=""

if( ans.count("@")>1):
    ok= True
    for i in range(len(ans)):
        if( ans[i]=='@' ):
            if( ok ) : 
                ans1 += '@'
                ok = False
            else :
                ans1 += "at"
        else :
            ans1 += ans[i]

else: ans1=ans

print(ans1)

File: Codeforces/C_Email_address.cpp
// #include <iostream>
#include<bits/stdc++.h>
using namespace std;


#define all(a) a.begin(),a.end()
#define fi first
#define se second
#define pb push_back
#define pyes cout<<"YES"<<endl
#define pno cout<<"NO"<<endl
#define ce cout<<endl
#define forr(i,n) for(int i=0 ; i<n ; i++)
#define pvec(x)  for(auto &a : x){cout<<a<<" ";}
#define pvec2d(v) for(int i=0 ; i<v.size() ; i++){for(int j=0 ; j<v[i].size() ; j++){cout<<v[i][j]<<" ";}cout<<endl;}

#define each(a, x) for (auto &a : x)
#define ll long long
#define MOD 1000000007;
#define test_cases(T) int T ; cin>>T ; while(T--)







int main() 
{

#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif 

    ios_base::sync_with_stdio(0); cin.tie(0);

    string s; cin >> s;

    string t = s;


    int n = s.size();

    ll int att = s.find("at");
    while (true && att < 1000) {

        s.replace(att, 2, "@");
        att = s.find("at");
        n = s.size();
        if (att >= n) break;
    }

    n = s.size();
    ll int dtt = s.find("dot");

    while (true && dtt < 1000) {

        s.replace(dtt, 3, ".");
        dtt = s.find("dot");
        n = s.size();
        if (dtt >= n) break;
    }

    string ans = "";

    if (s.find('.') == 0) {
        s.replace(0, 1, "dot");
    }
    if (s.rfind('.') == (s.size() - 1)) {
        s.replace(s.size() - 1, 1, "dot");
    }

    if (s.find('@') == 0) {
        s.replace(0, 1, "at");
    }

    if (s.rfind('@') == (s.size() - 1)) {
        s.replace(s.size() - 1, 2, "at");
    }

    cout << s;

}